programs/autocrat/src/instructions/finalize_proposal.rs

```diff
@@ -1,10 +1,12 @@
+use conditional_vault::{cpi::accounts::ResolveQuestion, ResolveQuestionArgs};
+
 use super::*;
 
 #[derive(Accounts)]
 pub struct FinalizeProposal<'info> {
     #[account(mut,
-        has_one = base_vault,
-        has_one = quote_vault,
+        // has_one = base_vault,
+        // has_one = quote_vault,
```

Comment:
shouldn't there be a `has_one = question` now?

Reply:
this might be a critical issue.
1. create proposal with question A, and let it pass
2. create question B with the same proposal as oracle
3. call finalize_proposal with the proposal, but with question B
4. question B will now be resolved, and the proposal will be blocked from resolving with question A because the state is now passed/failed

Reply:
yep, this was indeed a critical issue. resolved: e616e3a83ba43c6fc94cef4d71e53fdbc0658b4f